dashboard/lib/api/v1/school_autocomplete.rb

```diff
@@ -0,0 +1,54 @@
+require 'singleton'
+
+class SchoolAutocomplete
+  include Singleton
+
+  # The minimum query length.
+  MIN_QUERY_LENGTH = 4
+
+  # Queries the schools lookup table for schools that match the user-defined
+  # search criteria.
+  # @param query [String] the user-define query string
+  # @param limit [int] the maximum number of results to return
+  # @return [Array] an array of JSON formatted schools
+  def get_matches(query, limit)
+    return [] if query.length < MIN_QUERY_LENGTH
+    schools = School.
+      where("MATCH(name,city) AGAINST(? IN BOOLEAN MODE)", to_search_string(query)).
+      limit(limit)
+    results = schools.map do |school|
+      Serializer.new(school).attributes
+    end
+    return results
```

Comment:
Ruby tip: Ruby implicitly returns the last-evaluated expression as the result of a method, so you could remove `results = schools.map [...] return results` and just end the method with `schools.map` expression to simplify.